### ISSUE TITLE
Rambox (snap) won't start, apparmor denied.

### ISSUE THREAD

**Issue by klokop** [ramboxapp/community-edition#2494]
Rambox used to run fine (like,  2 weeks ago), but now it won't start.
This is what dmesg is telling me:

```
[Tue Nov 26 11:28:48 2019] audit: type=1400 audit(1574764129.739:113): apparmor="DENIED" operation="open" profile="snap.rambox.rambox" name="/media/SG6TB/Music/" pid=7748 comm="head" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
[Tue Nov 26 11:28:48 2019] audit: type=1400 audit(1574764129.747:114): apparmor="DENIED" operation="rmdir" profile="snap.rambox.rambox" name="/media/SG6TB/Music/" pid=7754 comm="rmdir" requested_mask="d" denied_mask="d" fsuid=1000 ouid=1000
```
I have ~/Music symlinked to the mount point listed above.

> Rambox: 0.7.2 (snap)

> OS: ubuntu bionic

> Arch: x64

**Comment by fvulich** [comment]
Could you try another package?

**Comment by klokop** [comment]
The snap package 'Postman' works fine, although it produces the same Apparmor warnings.

**Comment by klokop** [comment]
Last couple o' lines from 'strace rambox':

```
openat(AT_FDCWD, ".", O_RDONLY|O_NOFOLLOW|O_CLOEXEC|O_PATH|O_DIRECTORY) = 3
fstat(3, {st_mode=S_IFDIR|0750, st_size=4096, ...}) = 0
chdir("/")                              = 0
getresuid([1000], [1000], [1000])       = 0
getresgid([1000], [1000], [1000])       = 0
geteuid()                               = 1000
write(2, "need to run as root or suid", 27need to run as root or suid) = 27
write(2, "\n", 1
)                       = 1
exit_group(1)                           = ?
+++ exited with 1 +++
```
The 'need to run as root or suid' seems worrying.

**Comment by frederikmoellers** [comment]
I get the same strace on OpenSUSE Tumbleweed. The AppArmor lines don't appear in my dmesg or AppArmor logs, though.
On a second machine though, rambox runs just fine under the same distro. I will check if I can find out where the setups differ (maybe different versions of snapd, some snaps or some configurations).

**Comment by ric2b** [comment]
Same issue, won't start.
With `dmsg` I see
```
[ 4066.762111] audit: type=1400 audit(1575812839.979:875): apparmor="DENIED" operation="open" profile="snap.rambox.rambox" name="/mnt/StorageA/iTunes/Music/" pid=9864 comm="head" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
[ 4066.783118] audit: type=1400 audit(1575812840.003:876): apparmor="DENIED" operation="rmdir" profile="snap.rambox.rambox" name="/mnt/StorageA/iTunes/Music/" pid=9870 comm="rmdir" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
```

Not sure why it would try to access my music directory (which is on a separate drive that I share with Windows)

**Comment by ric2b** [comment]
Still happening with 0.7.3 (snap), by the way.
I'm using Ubuntu 19.10

**Comment by ric2b** [comment]
So I got tired of waiting and found a workaround [here](https://github.com/RocketChat/Rocket.Chat/issues/14562#issuecomment-498321790).

Basically, edit `/var/lib/snapd/apparmor/profiles/snap.rambox.rambox` and add a

**Comment by ric2b** [comment]
@fvulich Just noticed this happens to me and @klokop for our non-standard, outside `$HOME` Music directories.
I guess the last Rambox versions are trying to access the Music directory for some reason and if they're not in our Home directory it doesn't have permissions for it and crashes.

I just edited `~/.config/user-dirs.dirs` to change my Music directory, changed `$XDG_MUSIC_DIR`.

Hope it helps.

**Comment by fvulich** [comment]
Thanks for all the feedback

**Comment by ric2b** [comment]
This is fixed for me in version 0.7.7, after giving Rambox the new permission to access external storage devices.

**Comment by frederikmoellers** [comment]
Indeed, I haven't encountered this problem anymore. Imo we can close this issue.
